### DISCUSSION TITLE
Common single letter variables in the mapping community?

### DISCUSSION THREAD

**Discussion by Rowlando13** [NREL/mappymatch#99] [Q&A]
Are the following single letter variable names understood? I noticed them in some places in the code base: 

utils/plot.py
- m (context geofence)
- c (context geofence) 
- g (context coordinates / geofence)

**Comment by machallboyd** [comment]
Would argue that even if they are understood in the community, it's better to spell them out if it's not clear from the context of the code.

For instance:

```
i = np.argmax([m.distance for m in self.matches if m.road])
cutting_points.append(CuttingPoint(i))
```

`m` is clear for the entire scope of the variable. It's a temporary variable to hold values from `self.matches`.

`i` isn't as clear and would be better named `distance` or something similar, especially if it's used again in more places than just the following line.

**Reply by Rowlando13** [reply]
@nreinicke I agree in principle with @machallboyd . I just wanted some community feedback before I change put in the effort for the utils/plot.py

**Reply by nreinicke** [reply]
yeah I'm open to renaming the short variables in cases where it would make the code more readable. I agree with @machallboyd in distinguishing between variable scope. In the case of a list comprehension I would favor the short name to keep the syntax compact but we could definitely benefit from renaming those short variables that get used later (like `m` becoming `folium_map` or something descriptive like that)

**Reply by Rowlando13** [reply]
Agreed.

**Reply by Rowlando13** [reply]
Final answer is that they are not well know enough. Will rename them will paying attention to readability and scope.
